WIP: Export view

Neihtq opened the pull request:
- created richtext editor with ```draft-js```
- added inlinestyle features: italic, bold, underlined, strikethrough
- setup UI for inlinestyle features: EditorTools
- setup UI for Export-View componentes: RightSidepanel

thetommes committed:
Altes Modul entfernt (react-md)

thetommes committed:
dependency material-ui hinzugefügt

thetommes committed:
dev-dependeny webpack file-loader hinzugefügt.
Wir zum Laden von Bildern/Logos benötigt

thetommes committed:
Ordner "Assets hinzugefügt" (für Medien)

thetommes committed:
Webpack file-loader konfiguriert

thetommes committed:
Erster Header-Entwurf

thetommes committed:
Styling angepasst

thetommes committed:
Dynamic page-title added

thetommes committed:
Changed dynamic title

Neihtq committed:
creat Doc editor

Neihtq committed:
create constants file; add sapmle text

Neihtq committed:
create side panel

Neihtq committed:
column layout

Neihtq committed:
add side drawer, editor background changed to white

Neihtq committed:
create editortools section

Neihtq committed:
change color on font mode click

Neihtq committed:
Lifting state up

Neihtq committed:
lift editorState up

Neihtq committed:
implement toggle inline style

Neihtq committed:
update gitignore

Neihtq committed:
remove vscode folder

Neihtq committed:
resolve merge conflict

csehlke commented on EiDiA_frontend/src/components/ExportView/DocEditor.js:
```diff
@@ -0,0 +1,68 @@
+    { text: '64', value: 64 },
+    { text: '72', value: 72 },
+    { text: '90', value: 90 },
+  ];
```
ich hätte das in eine extra Datei ausgelagert, nachdem wir das ja mindestens 2x verwenden

csehlke commented on EiDiA_frontend/src/components/ExportView/DocEditor.js:
```diff
@@ -0,0 +1,68 @@
+            if (this.editor) {
+                this.editor.focus();
+            }
+        };
```
gibt es einen Grund warum die Funktionen hier im Construktor stehen und nicht außerhalb?

csehlke commented on EiDiA_frontend/src/components/ExportView/DocEditor.js:
```diff
@@ -0,0 +1,68 @@
+                <Editor
+                    ref={this.setEditor}
+                    editorState={editorState}
+                    onChange={onChange}
```
hätte die Konstanten direkt rein geschrieben, ist aber Geschmackssache

csehlke commented on EiDiA_frontend/src/components/ExportView/EditorTools.js:
```diff
@@ -0,0 +1,104 @@
+import FormatAlignCenterIcon from '@material-ui/icons/FormatAlignCenter';
+import FormatAlignRightIcon from '@material-ui/icons/FormatAlignRight';
+import FormatAlignJustifyIcon from '@material-ui/icons/FormatAlignJustify';
+import FormatUnderlinedIcon from '@material-ui/icons/FormatUnderlined';
```
ist die Frage ob wir hier auch gleich schon auf dei react-icons wechseln sollten

csehlke commented on EiDiA_frontend/src/components/ExportView/EditorTools.js:
```diff
@@ -0,0 +1,104 @@
+                <IconButton onMouseDown={() => this.handleClick("align riht")}>
+                    <FormatAlignRightIcon />
+                </IconButton>
+                <IconButton onMouseDown={() => this.logClick("alighn justify")}>
```
*align* -> Typo

csehlke replied:
und warum logClick und nicht handleClick?

csehlke commented on EiDiA_frontend/src/components/ExportView/EditorTools.js:
```diff
@@ -0,0 +1,104 @@
+                <IconButton onMouseDown={() => this.handleClick("Align center")}>
+                    <FormatAlignCenterIcon />
+                </IconButton>
+                <IconButton onMouseDown={() => this.handleClick("align riht")}>
```
*right* -> typo

csehlke commented on EiDiA_frontend/src/components/ExportView/RightSidepanel.js:
```diff
@@ -0,0 +1,45 @@
+  render() {
+    return (
+      <div style={styles.drawer}>
+          <div/>
```
gibt es einen Grund für dieses div?

csehlke commented on package.json:
```diff
@@ -27,17 +27,21 @@
     "webfontloader": "1.6.28"
   },
   "devDependencies": {
+    "@material-ui/icons": "^4.9.1",
```
statt denen die react-icons verwenden

csehlke commented on EiDiA_frontend/src/views/ExportView.js:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    onChange(editorState) {
+        this.setState({["editorState"]: editorState})
```
müsste das nicht `this.setState({editorState: editorState})` sein?

csehlke commented on EiDiA_frontend/src/views/ExportView.js:
```diff
@@ -0,0 +1,68 @@
+    toggleInlineStyle(style) {
+        const inlineStyle = style.toUpperCase();
+        this.setState({
+          ["editorState"]: RichUtils.toggleInlineStyle(this.state.editorState, inlineStyle)
```
müsste das nicht `this.setState({editorState: RichUtils.toggleInlineStyle(this.state.editorState, inlineStyle)})` sein?

csehlke commented on EiDiA_frontend/src/views/ExportView.js:
```diff
@@ -0,0 +1,68 @@
+                <Row>
+                    <Column>
+                        Sidebar
+                    </Column>
```
um die Sidebar brauchst du dich hier nicht kümmern, die kommt von 'oben'

csehlke commented on EiDiA_frontend/src/views/ExportView.js:
```diff
@@ -0,0 +1,68 @@
+                    </Column>
+                    <Column>
+                        <DocEditor 
+                            ref={(docEditor) => {this.docEditor = docEditor}}
```
ich hätte `ref={this.docEditor}` geschrieben, oder wird das an andere Stelle neu assigned

Neihtq replied:
Meine toggle inlinestyle funktion funktioniert sonst nicht mehr. Die referenz vom DocEditor brauch ich für diese Funktion

csehlke commented on EiDiA_frontend/src/views/ExportView.js:
```diff
@@ -0,0 +1,68 @@
+                    <Column>
+                        <DocEditor 
+                            ref={(docEditor) => {this.docEditor = docEditor}}
+                            editorState={editorState}
```
gleiche wieder mit der Constante hier

csehlke commented on EiDiA_frontend/src/components/ExportView/EditorTools.js:
```diff
@@ -0,0 +1,104 @@
+                    <FormatAlignLeftIcon />
+                </IconButton>
+                <IconButton onMouseDown={() => this.handleClick("Align center")}>
+                    <FormatAlignCenterIcon />
```
hier fehlen jeweils die style änderungen

Neihtq committed:
add blokstyle toggle

Neihtq committed:
docSearch: Search results and selected documents UI

Neihtq committed:
make rightside panel sub-components dynamic; create Export Section and docSearch component

Neihtq committed:
create Template Selection

Neihtq committed:
add datastructure for dynamic components

Neihtq committed:
implement change text on template select

Neihtq committed:
create FloatingWindow

Neihtq committed:
implement Dialog

Neihtq committed:
merge file cabinet to export-view branch

Neihtq committed:
dynmic variable list

Neihtq committed:
add Edit page

Neihtq committed:
create Variable value setter

Neihtq committed:
make header clickable and change view of export page

Neihtq committed:
implement set-values-for-variables-funciton

Neihtq committed:
manual variable mapping

Neihtq committed:
implement new datamodel for variables; working variable table; implement preview

Neihtq committed:
poper document mapping and path checking

Neihtq committed:
Make docsearch scrollable and fixed height

Neihtq committed:
light refactoring

Neihtq committed:
Merge brnach 'master' into 'export-view'

Neihtq committed:
merge master into export-view

Neihtq committed:
merge master into export view

Neihtq committed:
changed icons to react-icon; resolve for pull request

csehlke committed:
clean up using IDE tools
remove material icons again
remove anything that is not related to the export functionality and not yet part of master (i.e. file browser, record stuff)

csehlke commented on EiDiA_frontend/src/support files/constants.js:
```diff
@@ -0,0 +1,40 @@
+    }
+};
+
+export const fontSizes = [
```
unused code

csehlke commented on EiDiA_frontend/src/views/ExportView.js:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    componentDidMount() {
+        this.props.setTitle(this.state.currentPage);
```
ich hätte hier als Hauptüberschrift "Create Document" gesetzt, denn die andere Info steckt ja in der Unterüberschrift

csehlke commented on EiDiA_frontend/src/views/ExportView.js:
```diff
@@ -0,0 +1,34 @@
+    }
+    changeView(page) {
+        this.props.setTitle(page);
+        this.setState({currentPage: page, readOnly: (page=="Select Template")});
```
Vergleiche immer mit 3 = siehe #65

csehlke commented on EiDiA_frontend/src/components/ExportView/Header.js:
```diff
@@ -0,0 +1,62 @@
+
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
```
lots of unused code L 34-43

csehlke commented on EiDiA_frontend/src/components/ExportView/Header.js:
```diff
@@ -0,0 +1,62 @@
+        <div className={classes.root}>
+                <Toolbar disableGutters={true} className={classes.toolBar}>
+                    <Typography variant="subtitle1" align="center" className={props.title == "Select Template" ? classes.title : classes.titleUnselect}>
+                        <div onClick={() => props.changeView("Select Template")}>Select Template</div>
```
Gibt es einen Grund warum das "onClick" nicht schon im Typography drin ist?

csehlke commented on EiDiA_frontend/src/components/ExportView/Header.js:
```diff
@@ -0,0 +1,62 @@
+        flexGrow: 1,
+        color: "black",
+        background: "#b0b0b0",
+        //boxShadow: "0px 0px 4px 0px rgba(0,0,0,0.75)"
```
unused code

csehlke commented on EiDiA_frontend/src/components/ExportView/Header.js:
```diff
@@ -0,0 +1,62 @@
+        minHeight: "10px"
+    },
+    menuButton: {
+        marginRight: theme.spacing(2)
```
menueButton wird nie genutzt

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+
+function fetchDocumentData(docList) {
+    const documentData = documentMockData;
+    return documentData;
```
add a TODO if not implemented

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+export default class ExportMainView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.editorText = llorem["Template 0"];
```
ist das nicht auch Teil vom State?

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+
+        this.docEditor = React.createRef();
+
+        this.actionSet = {
```
magst du hier kommentieren was das ist?

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+    return /^(?:\/|[a-z]+:\/\/)/.test(string);
+}
+
+const components = {
```
magst du hier kommentieren was das ist?

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+            selectedTemplate: null,
+            open: false,
+            variables: {},
+            selectedDocs: [],
```
was sicher gut wäre, hier Beispielausprägungen zu kommentieren

csehlke replied:
für sowohl variables als auch selecetedDocs

csehlke commented:
Warning: Each child in a list should have a unique "key" prop.
Check the render method of `SetValueSection`. See https://fb.me/react-warning-keys for more information.
    in WithStyles(ForwardRef(TableRow)) (created by SetValueSection)
    in SetValueSection (created by RightSidePanel)
    in div (created by RightSidePanel)
    in RightSidePanel (created by ExportMainView)
    in div (created by styled.div)
    in styled.div (created by ExportMainView)
    in div (created by styled.div)
    in styled.div (created by ExportMainView)
    in div (created by ExportMainView)
    in ExportMainView (created by ExportView)

csehlke commented:
Uncaught TypeError: documentData[documentIndex] is undefined
    mapValues ExportMainView.js:154
    componentDidUpdate ExportMainView.js:268

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+        var variables = this.state.variables;
+        var selectedDocs = this.state.selectedDocs;
+        var documentData = []
+        selectedDocs.forEach((name) => { if (name in documents) documentData.push(documents[name])});
```
was passiert hier?
falls hier etwas identifiziert werden soll, warum nicht by Id sonder by Name?

Neihtq replied:
```name``` zu ```docName``` 
weil selectedDocs Dokumentennamen enthält und nach denen identifiziert wird

csehlke commented:
ch glaube du musst im "Edit" verbieten im Dokument Änderungen im Text selber zu machen, das mach dir den Text kaputt, wenn du dannach Variablen ersetzt

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+        var editorText = this.getTextFromEditorstate(newState.editorState);;
+        for (let k of Object.keys(variables)) {
+            if (isPath(k.slice(1))) {
+                let tmp = k.split("/");
```
tmp ist ein schlechter Name, wenn das ein anderer nochmal verstehen soll... tmp-was

csehlke replied:
und auch variable und value sind auch nur sinnvoll, wenn es tatsächlich um die Variablen im Document geht

Neihtq replied:
umbenannt zu
```
let variableTokens
let docVariable
let indexString
let documentIndex
let indices
let docValue
```

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+        this.setState(newState);
+    }
+
+    addToList(element) {
```
addSelectedDocumentToList

csehlke commented:
was hältst du davon die drei Seitennamen als Constanten zu declarieren?

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+        let newVariableState = {}
+        for (let k of Object.keys(newVariables)) {
+            let index = (k in currVariables) ? currVariables[k]["index"] : newVariables[k]["index"];
+            let value = (k in currVariables) ? currVariables[k]["value"] : newVariables[k]["value"];
```
ist geschmackssache, aber hier müsste ja jeweils auch `newVariables[k].value` gehen nachdem wir erst ein Array und dann ein Object haben

Neihtq replied:
struktur nochmal grundlegend geändert

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+    // retirm objects with index and set value and variable as key
+    extractVariables(editorState) {
+        const value = this.getTextFromEditorstate(editorState);
+        const arr = value.split(" ");
```
value und arr, es gibt sicher sprechendere Namen

Neihtq replied:
umbenannt:
```
const value --> const editorText
const arr --> tokenizedText
```

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+
+    componentDidUpdate(prevProps) {
+        if (this.props.currentPage !== prevProps.currentPage) {
+            if (this.props.currentPage == "Edit") this.mapValues();
```
```suggestion
        if (this.props.currentPage !== prevProps.currentPage && this.props.currentPage == "Edit") {
            this.mapValues();
```

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+            editorState: EditorState.createWithContent(ContentState.createFromText(this.editorText)),
+            textAlignment: "left",
+            selectedTemplate: null,
+            open: false,
```
What is open? Gibt es keinen sprechenderen Namen dafür?

Neihtq replied:
umbenannt zu ```showDialog```: ne bool flag die entscheidet ob der Dialog/FloatingWindow angezeigt wird oder nicht (Save Template, Export Document)

material-ui's Dialog component verwendet die open flag für diese Funktion, von daher in der FloatingWindow class folgendes drin:
```<Dialog ... open={this.props.showDialog}> ```

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+    "Edit": {
+        comp1: "editorTools",
+        comp2: "docSearch",
+        comp3: "setValueSection"
```
Wenn man an dieser Stelle schon die Componenten einsetzte, spart man sich später das Mapping im RightSidepanel

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+            <div>
+                <Row>
+                    <Column>
+                    </Column>
```
Spricht was degegen nur 2 Spalten zu machen?

Neihtq replied:
nicht wirklich ... wollte links bisschen frei fläche haben, damit der editor mittig ist.

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+            selectedDocs={props.selectedDocs}
+            download={props.download}
+            editorState={props.editorState}
+        />);
```
inline wahrscheinlich die schönere Lösung: also
```
{props.currentPage === "Select Template" ? "" : <FloatingWindow.../>}
```

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/DocEditor.js:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    setEditor(editor) {
+        this.editor = editor;
```
ist Editor kein State?

Neihtq replied:
angepasst, war copy&paste von vorherigen versionen

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+                        <RightSidepanel
+                            comp1={componentSet.comp1}
+                            comp2={componentSet.comp2}
+                            comp3={componentSet.comp3}
```
warum nicht einfach das gesamte Object `componentSet` übergeben?

csehlke commented on EiDiA_frontend/src/components/ExportView/ExportMainView.js:
```diff
@@ -0,0 +1,345 @@
+                            onAction2_2={actionSet.onAction2_2 || null}
+                            onAction3_1={actionSet.onAction3_1 || null}
+                            onAction3_2={actionSet.onAction3_2 || null}
+                            onAction3_3={actionSet.onAction3_3 || null}
```
und dann schon bei der deklaration die dritte action=null setzten

Neihtq replied:
wie meinst du das? auf der Edit view hat die dritte sub component 3 actions

csehlke commented on EiDiA_frontend/src/components/ExportView/FloatingWindow.js:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    updateSearch(event){
+        this.setState({search: event.target.value.substr(0,20)});
```
warum begranzt auf 20 Zeichen?

Neihtq replied:
war glaub ich von philipps Filecabint damals übernommen. Aber da es nicht fix ist, ob beim export die dokumente in einem Record gespeichert werden und eig die templates global gespeichert werden, habe ich das file cabinet aus dem window gelöscht und damit auch diese funktion.

csehlke commented on EiDiA_frontend/src/components/ExportView/FloatingWindow.js:
```diff
@@ -0,0 +1,59 @@
+            }
+        );
+        return(
+            <Dialog onClose={this.props.onClose} fullWidth={true} maxWidth={'lg'} id="temp" aria-labelledby="simple-dialog-title" open={this.props.open}>
```
warum `id="temp" aria-labelledby="simple-dialog-title"`

Neihtq replied:
war copy paste, habs gelöscht

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,131 @@
+}
+
+const FlexRow = styled.div`
+    margin: 0 5% 0 5%;
```
`margin: 0 5%`
ist identisch

Neihtq replied:
FlexRow entfernt

csehlke commented:
in den DialogWindows fehlt die Möglichkeit einen Record aus zu wählen => wollen wir die überhaupt einem Record zuordnen oder zumindest die Templates global speichern?

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,131 @@
+export default class ExportDocumentWindow extends React.Component {
+    render() {
+        let filteredRecords = this.props.filteredRecords;
+        let selectedDocs = this.props.selectedDocs;
```
die Variablen brauchts nicht

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/Preview.js:
```diff
@@ -0,0 +1,36 @@
+export default class Preview extends React.Component {
+    
+    render() {
+        let editorState = this.props.editorState;
```
variable ohne Mehrwert

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,131 @@
+                        Used Documents
+                    </Typography>
+                    <FormGroup>
+                        {selectedDocs.map((doc) => <FormControlLabel control={<Checkbox color="primary"/>} label={doc}/>)}
```
die Variable müsste eher usedDocs statt selectedDocs heißen, weil ausgewählt habe ich sie ja noch nicht

wird auch irgendwo gespeichert welche Dokumente ich im Dialog ausgewählt habe?

Neihtq replied:
habs geändert zu ```usedDocs```

im Dialog ausgewählte Dokumente werden jetzt im state gespeichert.

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,131 @@
+                    <Input placeholder="Template Name"  inputProps={{ 'aria-label': 'description' }} />
+                    </Row>
+                    <Row>
+                        <div width="80%">
```
es gibt kein width attribute auf einem div, das muss in ein style eingepackt werden

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,131 @@
+                            <Typography variant="subtitle2">
+                                Template Name
+                            </Typography>
+                            <Input placeholder="Search Records ..." fullWidth={true}  inputProps={{ 'aria-label': 'description' }} value={this.props.value} onChange={this.props.updateSearch.bind(this)} />
```
description ist eher nicht das richtige Label

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,131 @@
+                            disableElevation
+                        >
+                            Send as E-Mail
+                        </Button>
```
ale die Funktionen die wir nicht implementieren sollten wir vielleicht disablen

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,131 @@
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    height: 50%;
```
height 50% funktionier nicht ganz, evtl klappt max-height

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/SaveTemplateWindow.js:
```diff
@@ -0,0 +1,95 @@
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    height: 50%;
```
same here wie im anderen ExportDocumentWindow

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/SaveTemplateWindow.js:
```diff
@@ -0,0 +1,95 @@
+                            Created Template (Preview)
+                        </Typography>
+                        <Box style={styles.box} component="span" display="block" p={1} m={1} bgcolor="LightGray">
+                        </Box>
```
die Box hier ähnlich wie im anderen Window mit der Preview ersetzten

csehlke commented on EiDiA_frontend/src/components/ExportView/Subcomponents/SaveTemplateWindow.js:
```diff
@@ -0,0 +1,95 @@
+                        <Input placeholder="Template Name"  inputProps={{ 'aria-label': 'description' }} />
+                        </Row>
+                        <Row>
+                            <div width="80%">
```
Width auf div geht nicht

csehlke commented on EiDiA_frontend/src/components/ExportView/styles.css:
```diff
@@ -0,0 +1,58 @@
+    color: cyan;
+    cursor: pointer;
+  }
+  
```
wird die Datei irgendwo verwendet ich habe keinen Import gefunden?

Neihtq replied:
hab die datei gelöscht

csehlke reviewed (changes requested):
Hab bis jetzt die Hälfte angeschaut, der Rest kommt später

Neihtq committed:
Add case with 0 selected documents

Neihtq committed:
merge export-view branch

Neihtq committed:
Merge pull request #65 from csehlke/review/exportView

Clean up export view for review

Neihtq committed:
Handle duplicate variable

Neihtq committed:
disable document editing in Edit view

Neihtq committed:
add preview for save template floating window

Neihtq committed:
implement get List request

Neihtq committed:
implement get template content request

Neihtq committed:
create endpoint for docSearch

Neihtq committed:
backend: proper return objects

Neihtq committed:
create endpoint for getDocument data

Neihtq committed:
handle linebreaks for variable scanning

Neihtq committed:
create endpoint for document export/download

Neihtq committed:
create PDF downloader, creator and view

Neihtq committed:
restructured floatingwindow / dialog

Neihtq committed:
clean up

Neihtq committed:
finish clean up

Neihtq committed:
fix wrong string in constants class

Neihtq committed:
change type of selected documents to object {name, id}

Neihtq committed:
clean up; creat helper methods

Neihtq committed:
merge export-view to master

Neihtq committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	.gitignore
#	EiDiA_frontend/src/App.js
#	EiDiA_frontend/src/components/Page.js
#	EiDiA_frontend/src/components/record/RecordSymbol.js
#	EiDiA_frontend/src/components/search/SearchResults.js
#	EiDiA_frontend/src/views/FileCabinetView.js
#	EiDiA_frontend/src/views/RecordView.js
#	package-lock.json
#	package.json

Neihtq committed:
avoid scrollbary on parent level; use HttpService

Neihtq committed:
remove utils.js

csehlke commented on EiDiA_backend/src/controllers/exportTemplate.js:
```diff
@@ -1,17 +1,27 @@
+                {name: "Template 2", id: "t_2"},
+            ];
+
+            res.status(200).json({response: dummyData});
```
```suggestion
            res.status(200).json({exportTemplates: dummyData});
```

csehlke commented on EiDiA_backend/src/controllers/exportTemplate.js:
```diff
@@ -21,21 +31,64 @@ const listTemplates = (req, res) => {
+    }
+
+    res.status(200).json({response: out});
 };
```
warum nicht wie hier #59

csehlke commented on EiDiA_backend/src/routes/exportTemplate.js:
```diff
@@ -7,8 +7,10 @@ const middleWares = require('../middlewares');
-router.get('/exportDocument', middleWares.checkAuthentication, ExportTemplateController.exportDocument);
-router.get('/download', middleWares.checkAuthentication, ExportTemplateController.download);
+router.get('/export/:documents', middleWares.checkAuthentication, ExportTemplateController.exportDocument);
+router.get('/search/:query', middleWares.checkAuthentication, ExportTemplateController.search);
```
warum nicht wie in #59

csehlke commented on EiDiA_frontend/src/support files/constants.js:
```diff
@@ -0,0 +1,27 @@
+
+export const Column = styled.div`
+    flex: 50%;
+`;
```
magst du das in das in die Datei vom Philipp verschieben?

csehlke commented on EiDiA_frontend/src/support files/constants.js:
```diff
@@ -0,0 +1,27 @@
+    flex: 50%;
+`;
+
+const BASE_URL = "http://localhost:3000/";
```
das gibt es schon im constants.js im root folder

csehlke commented on EiDiA_frontend/src/support files/constants.js:
```diff
@@ -0,0 +1,27 @@
+    selectTemplate: "Select Template",
+    editTemplate: "Edit Template",
+    edit: "Edit"
+}
```
das gehört näher zur Export View, evtl ist das gut aufgehoben in ExportView.js

csehlke replied:
todo

csehlke commented on EiDiA_frontend/src/support files/constants.js:
```diff
@@ -0,0 +1,27 @@
+    getDocs: BASE_URL + "exporttemplate/documents/",
+    saveTemplate: BASE_URL + "exporttemplate/save/",
+    exportDocs: BASE_URL + "exporttemplate/export/"
+}
```
und das hier hätte ich in deinen Backend Controler geschrieben, oder wird das noch wo anders verwendet? Dann vielleicht für alle einheitlich herausziehen und nach constants.js im root folder

Neihtq committed:
fix addSelectedDocumentToList

Neihtq committed:
delete unnecessary buttons

Neihtq committed:
remove unnecessary lines

csehlke commented on EiDiA_frontend/src/components/filetable/FileTable.js:
```diff
@@ -0,0 +1,18 @@
+            </div>
+        );
+    }
+}
```
weg damit, der Filetable ist inzwischen in den record folder gewandert

csehlke commented on EiDiA_frontend/src/components/exportView/Header.js:
```diff
@@ -0,0 +1,57 @@
+                            className={props.title === pageNames.selectTemplate ? classes.title : classes.titleUnselect}
+                            onClick={() => props.changeView(pageNames.selectTemplate)}
+                >
+                    Select Template
```
use the constants

csehlke commented on EiDiA_frontend/src/components/exportView/Header.js:
```diff
@@ -0,0 +1,57 @@
+                            className={props.title === pageNames.editTemplate ? classes.title : classes.titleUnselect}
+                            onClick={() => props.changeView(pageNames.editTemplate)}
+                >
+                    Edit Template
```
same here

csehlke commented on EiDiA_frontend/src/components/exportView/Header.js:
```diff
@@ -0,0 +1,57 @@
+                            className={props.title === pageNames.edit ? classes.title : classes.titleUnselect}
+                            onClick={() => props.changeView(pageNames.edit)}
+                >
+                    Edit
```
same

csehlke commented on EiDiA_frontend/src/components/exportView/Header.js:
```diff
@@ -0,0 +1,57 @@
+    }
+));
+
+const MenuAppBar = (props) => {
```
ist MenuAppBar der richtige Name? es geht ja nicht um die App sindern um die ExportView => `ExportMenuBar`?

csehlke committed:
fix package-lock npm vulnerabilities

csehlke commented on EiDiA_frontend/src/components/exportView/ExportMainView.js:
```diff
@@ -0,0 +1,417 @@
+            />}
+        </React.Fragment>
+    )
+}
```
Warum nicht inline?

csehlke commented on EiDiA_frontend/src/components/exportView/ExportMainView.js:
```diff
@@ -0,0 +1,417 @@
+    componentDidMount() {
+        HttpService.get(endpoints.getTemplateList, (resp) => {
+            let initTemplate = resp.response[0]
+            this.selectTemplate(initTemplate.name, initTemplate.id);
```
was passiert wenn du ein `response: []` vom backend bekommst, oder ist das irgendwo anders abgefangen?

csehlke reviewed (changes requested):
erste Drittel, mehr folgt

csehlke commented on EiDiA_frontend/src/components/exportView/ExportMainView.js:
```diff
@@ -0,0 +1,417 @@
+                            onAction2_1={actionSet.onAction2_1}
+                            onAction3_1={actionSet.onAction3_1}
+                            onAction3_2={actionSet.onAction3_2}
+                            onAction3_3={actionSet.onAction3_3}
```
warum das actionSet nicht ähnlich zum componentSet?

csehlke commented on EiDiA_frontend/src/components/HeaderExport.js:
```diff
@@ -0,0 +1,106 @@
+"use strict";
```
ist die Datei nicht an der falschen Stelle? ich häte die in den Ordner exportView verschoben

csehlke replied:
todo

csehlke commented on EiDiA_frontend/src/components/HeaderExport.js:
```diff
@@ -0,0 +1,106 @@
+    }
+}));
+
+const MenuAppBar = (props) => {
```
?? der Dateiname ist HeaderExport aber die Component heißt MenuAppBar?

csehlke replied:
-> wird nicht mehr genutzt -> entfernen

csehlke replied:
todo

csehlke replied:
todo

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/SaveTemplateWindow.js:
```diff
@@ -0,0 +1,54 @@
+                    </Typography>
+                </Row>
+                <Row>
+                    <Input placeholder="Template Name" inputProps={{'aria-label': 'description'}}/>
```
```suggestion
                    <Input placeholder="Template Name"/>
```
`description` ist keine Beschreibung

csehlke replied:
und vielleicht irgendeinen Hinweis in der GUI darstellen: not implemented, nachdem es ja noch nicht tut, da ua die Eingabe nirgends gespeichert wird

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/SaveTemplateWindow.js:
```diff
@@ -0,0 +1,54 @@
+    }
+}
+
+export default class SaveTemplateSection extends React.Component {
```
Filename und componet name stimmen nicht überein

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/SaveTemplateSection.js:
```diff
@@ -0,0 +1,24 @@
+export default class SaveTemplateSection extends React.Component {
+    render() {
+        return (
+            <div>
```
brauchts das surrounding div?

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,87 @@
+    }
+
+    render() {
+        let usedDocs = this.props.selectedDocs;
```
warum nicht inline?

csehlke replied:
und das prop zu usedDocs umbenennen, oder?

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,87 @@
+                        </Typography>
+                    </Row>
+                    <Row>
+                        <Input placeholder="Template Name"/>
```
Für was gebe ich hier einen Namen ein, wenn der nirgends weiter verwendet wird?

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,87 @@
+                    <FormGroup>
+                        {usedDocs.map((doc) => <FormControlLabel
+                            control={<Checkbox color="primary" onChange={() => this.selectDocument(doc)}/>}
+                            label={doc}/>)}
```
hier fehlt irgendwie der value für die checkbox, oder?

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/ExportDocumentWindow.js:
```diff
@@ -0,0 +1,87 @@
+
+    selectDocument(doc) {
+        let newState = this.state;
+        newState.selectedDocs.push(doc);
```
heißt ich kann ein Document nicht mehr aus der Liste holen und mt "select -> unselect -> select" zwei mal in die Liste bringen

csehlke commented:
Variablen gehen verloren wenn:
select template -> move to "Edit" -> replace one variable -> move back to edit Template (variable is gone) -> move to "Edit" (variable cannot be changed anymore")

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/SetValueSection.js:
```diff
@@ -0,0 +1,111 @@
+                        disableElevation
+                        onClick={this.props.onAction1}
+                    >Export</Button>
+                </div>
```
zum einen warum ein zusätzliches Div
zum anderen warum ist das nicht auch eine einzelne Componente wie bei den anderen Ansichten auch

Neihtq replied:
zusätzliche div gelöscht,

du meinst ein extra component für den Button? Bei Edit TemplateView wird die dritte/unterste Component aus der variablenliste und dem button bestehen (änderung ist im anderne branch drin). Nur bei Select Template besteht die dritte component aus den zwei Buttons. Generell sind die Components in den drei Flächen des RightSidePanels unterteils.

csehlke commented:
es ist sicher sinnvoll die Ordnerstruktur etwas anzu passen:
- components
  - export
    - dialog
      - everything to the dialgs
    - edit
      - everything to the right Side for edit
    - editTemplate
      - everything to the right Side for edit template
    - selectTemplate
      - everything to the right Side for select template
    - everything else
  - ...

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/TemplateList.js:
```diff
@@ -0,0 +1,64 @@
+                this.setState(newState);
+            },
+            (err) => {
+                console.log(error);
```
error statt err -> !

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/DocSearch.js:
```diff
@@ -0,0 +1,82 @@
+        this.search = this.search.bind(this);
+    }
+
+    search(e) {
```
hast du einen schöneren Namen für e?

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/EditorTools.js:
```diff
@@ -0,0 +1,110 @@
+        for (let key in newState.alignments) {
+            newState.alignments[key] = false
+        }
+        newState.alignments[alignmentStyle] = newFlag;
```
```suggestion
        newState.alignments[alignmentStyle] = true;
```

für alignment braucht man eine andere Strategie, sonst haben wir am ende überhaupt kein alignment ausgewählt

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/EditorTools.js:
```diff
@@ -0,0 +1,110 @@
+    }
+
+    setAlignment(e, alignmentStyle) {
+        e.preventDefault()
```
aus `e` macht `event` ;)

wann braucht man das `preventDefault()`, evtl muss man das überall sonst noch nachtragen

Neihtq replied:
preventDefault() ist notwendig wenn man inlinestyle oder alignmentstyle über den button click ändert während der editor fokussiert war. Der Fokus geht ohne preventDefault() verloren und die ausgewählten Styleänderungen treten nicht ein.

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/EditorTools.js:
```diff
@@ -0,0 +1,110 @@
+        this.setAlignment = this.setAlignment.bind(this);
+    }
+
+    setInlineStyle(e, inlineStyle) {
```
aus e macht event ;)

csehlke replied:
gilt auch für den Rest der datei

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/VariableList.js:
```diff
@@ -0,0 +1,25 @@
+                     p={1}
+                     m={1}
+                     bgcolor="LightGray">
+                    {variableKeys.map((key) => <li key={key}>{key}</li>)}
```
weißt du ob s für die li elemente eine Möglichkeit gibt die Punkte zu entfernen? Denn aktuell hast du abgeschnittene Punkte in der UI

Neihtq replied:
Ich hab keine abgeschnittenen Punkte bei mir. Kannst du mir ein Screenshot zeigen? Außerdem hab ich in nem anderen Branch die Breite vom Rightside panel und von der Variablenliste fixiert. Die momentane Größenregelung könnte mit den abgeschnittenen Punkten bei dir zusammenhängen.

csehlke replied:
![grafik](https://user-images.githubusercontent.com/29836078/87848353-c82ee480-c8df-11ea-8c0d-0528da78b622.png)

Neihtq replied:
hab margin gesetzt damit die li elemente weiter rechts sind

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/SetValueSection.js:
```diff
@@ -0,0 +1,111 @@
+    return {variable, value, source};
+}
+
+function convertData(variables, value = null) {
```
```suggestion
function convertData(variables) {
```

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/SetValueSection.js:
```diff
@@ -0,0 +1,111 @@
+        const rows = convertData(variableState);
+        return (
+            <div>
+                <FormControl style={{margin: "5px"}}>
```
```suggestion
                <FormControl style={{margin: "5px", width: '100%', padding: '0 10px 0 0'}}>
```
fixes the dropdown's width

csehlke commented on EiDiA_frontend/src/components/exportView/ExportMainView.js:
```diff
@@ -0,0 +1,417 @@
+        let variableState = newState.variables;
+        let index = variableState[newState.selectedVariable].index;
+        variableState[newState.selectedVariable].value = value;
+        variableState[newState.selectedVariable].source = value;
```
```suggestion
        variableState[newState.selectedVariable].source = 'user';
```
make it obvious that the user entered this value

csehlke commented on EiDiA_frontend/src/components/exportView/subcomponents/SetValueSection.js:
```diff
@@ -0,0 +1,111 @@
+                <IconButton style={{margin: "5px"}} color="primary"
+                            onClick={this.setValueToVariable}><MdAdd/></IconButton>
+                <TableContainer component={Paper}>
+                    <Table aria-label="simple table">
```
```suggestion
                    <Table>
```

Neihtq committed:
Update EiDiA_backend/src/controllers/exportTemplate.js

Co-authored-by: csehlke <csehlke@users.noreply.github.com>

Neihtq committed:
Update EiDiA_frontend/src/components/exportView/subcomponents/SaveTemplateWindow.js

Co-authored-by: csehlke <csehlke@users.noreply.github.com>

Neihtq committed:
Update EiDiA_frontend/src/components/exportView/subcomponents/SetValueSection.js

Co-authored-by: csehlke <csehlke@users.noreply.github.com>

Neihtq committed:
Update EiDiA_frontend/src/components/exportView/subcomponents/SetValueSection.js

Co-authored-by: csehlke <csehlke@users.noreply.github.com>

Neihtq committed:
Update EiDiA_frontend/src/components/exportView/subcomponents/SetValueSection.js

Co-authored-by: csehlke <csehlke@users.noreply.github.com>

Neihtq committed:
Update EiDiA_frontend/src/components/exportView/ExportMainView.js

Co-authored-by: csehlke <csehlke@users.noreply.github.com>

Neihtq committed:
applay changes from pr

Neihtq committed:
Merge branch 'master' into export-view
merge master to export-view

Neihtq committed:
creater Service for ExportView backend calls

Neihtq committed:
apply code review changes

Neihtq committed:
apply code review changes; introduce Alert for unsaved tempalte

Neihtq committed:
new file strucutre; removable linked documents; removable selected documents in docSearch

Neihtq committed:
fix RightSidePanel width

Neihtq committed:
implement alert when leaving edit view

Neihtq committed:
create new Template function

Neihtq committed:
add notification for unchanged variabls

Neihtq committed:
rename ExportSection to ButtonArea

Neihtq committed:
Merge branch 'master' into export-view

csehlke commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "python.linting.enabled": true
+}
```
gitignore

csehlke commented on EiDiA_frontend/src/components/export/selectTemplate/TemplateList.js:
```diff
@@ -0,0 +1,56 @@
+        ExportService.getAllTemplates().then((data) => {
+            newState.templateList = data.exportTemplates;
+            this.setState(newState);
+        })
```
es graucht auch ein catch
```suggestion
        })
        .catch(error => console.log(error));
```

Neihtq committed:
Update EiDiA_frontend/src/components/export/selectTemplate/TemplateList.js

Co-authored-by: csehlke <csehlke@users.noreply.github.com>

Neihtq committed:
apply changes from code review

Neihtq committed:
variableList styling

Neihtq committed:
delete constants file

csehlke committed:
remove second constants definition

csehlke committed:
Merge branch 'master' into export-view
